sdk/core/Azure.Core/tests/HttpPipelineFunctionalTests.cs

```diff
@@ -436,6 +437,166 @@ public async Task SendMultipartformData()
+            patchReq.Content = RequestContent.Create(Encoding.UTF8.GetBytes(patchBody));
+            changeset.Add(new RequestRequestContent(patchReq), new Dictionary<string, string> { { HttpHeader.Names.ContentType, "application/http" }, { cteHeaderName, Binary } });
+
+            request.Content = content;
+            using Response response = await ExecuteRequest(request, httpPipeline);
+            Console.WriteLine(requestBody);
+
+            if (_transportType == typeof(HttpClientTransport))
+            {
+                Assert.That(requestBody, Is.EqualTo($"--batch_{batchGuid}\r\n" +
+                    $"{HttpHeader.Names.ContentType}: multipart/mixed; boundary=changeset_{changesetGuid}\r\n" +
+                    $"\r\n" +
+                    $"--changeset_{changesetGuid}\r\n" +
+                    $"{HttpHeader.Names.ContentType}: application/http\r\n" +
+                    $"{cteHeaderName}: {Binary}\r\n" +
+                    $"\r\n" +
+                    $"POST {postUri} HTTP/1.1\r\n" +
+                    $"{HttpHeader.Names.Host}: {Host}\r\n" +
+                    $"{HttpHeader.Names.Accept}: {ApplicationJson}\r\n" +
+                    $"{DataServiceVersion}: {Three0}\r\n" +
+                    $"{HttpHeader.Names.ContentType}: {ApplicationJsonOdata}\r\n" +
+                    $"\r\n" +
+                    $"{post1Body}\r\n" +
+                    $"--changeset_{changesetGuid}\r\n" +
+                    $"{HttpHeader.Names.ContentType}: application/http\r\n" +
+                    $"{cteHeaderName}: {Binary}\r\n" +
+                    $"\r\n" +
+                    $"POST {postUri} HTTP/1.1\r\n" +
+                    $"{HttpHeader.Names.Host}: {Host}\r\n" +
+                    $"{HttpHeader.Names.Accept}: {ApplicationJson}\r\n" +
+                    $"{DataServiceVersion}: {Three0}\r\n" +
+                    $"{HttpHeader.Names.ContentType}: {ApplicationJsonOdata}\r\n" +
+                    $"\r\n" +
+                    $"{post2Body}\r\n" +
+                    $"--changeset_{changesetGuid}\r\n" +
+                    $"{HttpHeader.Names.ContentType}: application/http\r\n" +
+                    $"{cteHeaderName}: {Binary}\r\n" +
+                    $"\r\n" +
+                    $"PATCH {mergeUri} HTTP/1.1\r\n" +
+                    $"{HttpHeader.Names.Host}: {Host}\r\n" +
+                    $"{HttpHeader.Names.Accept}: {ApplicationJson}\r\n" +
+                    $"{DataServiceVersion}: {Three0}\r\n" +
+                    $"{HttpHeader.Names.ContentType}: {ApplicationJsonOdata}\r\n" +
+                    $"\r\n" +
+                    $"{patchBody}\r\n" +
+                    $"--changeset_{changesetGuid}--\r\n" +
+                    $"\r\n" +
+                    $"--batch_{batchGuid}--\r\n" +
+                    $""));
+            }
+            else //header ordering is different for HttpWebRequestTransport
```

Comment:
@pakrym FYI - any concerns about this behavior change?

Reply:
Hm, I don't think we should be using transport requests at all, instead, we should have something like MemoryRequest that always works the same and doesn't waste extra resources.